packages/api/src/share_link/service.ts

```diff
@@ -69,4 +73,120 @@ const createShareLink = async (
   };
 };
 
-export const shareLinkService = { createShareLink };
+const updatedShareLinkWouldBeUnlisted = (
+  updateRequest: UpdateShareLinkRequest,
+  currentShareLink: ShareLink
+): boolean =>
+  updateRequest.accessRestrictions === "none" ||
+  (currentShareLink.accessRestrictions === "none" &&
+    updateRequest.accessRestrictions === undefined);
+
+const postUpdatePermissionsLevel = (
+  updateRequest: UpdateShareLinkRequest,
+  currentShareLink: ShareLink
+): "contributor" | "editor" | "manager" | "owner" | "viewer" =>
```

Comment:
You know me -- I think it's clearer to have this kinda thing be in a named alias (or an enum) jfrom a separation of code responsibilities PoV (e.g. this line should just be saying what type of thing is returned; I wouldn't expect the postUpdatePermissionsLevel function definition to also be the place where we define the complete set of PermissionLevels)

Side note: if the linter doesn't like a thing that you want to be able to do (union of strings) I for one am OK with changing it to allow!

Reply:
As it happens, I know that an updated version of the linter switches its defaults to allow aliases for these. I think, as a separate PR, I'll make that update, and then standardize us on aliased unions for these sorts of things.